### PR TITLE
Split SummarizeContent from SummarizeSidebar

### PR DESCRIPTION
**Pull request by oisincoveney**
Moves the content of `SummarizeSidebar` into a separate component that will be used in #44494. We don't want to use the full sidebar logic and styling so moving the content will allow us more flexibility in the SDK.

### CODE REVIEW

**Review by deniskaber — COMMENTED** [review]
Hey @oisincoveney . Based on changes on this PR, it seems we don't get any benefits from such decomposition. 

Do you have plans to use `SummarizeContent` in some other place?
If not, then I think we don't need this component, and we should put it's content back to `SummarizeSidebar`. 

Hook is good, btw.

**Comment by oisincoveney** [comment]
> Hey @oisincoveney . Based on changes on this PR, it seems we don't get any benefits from such decomposition. 
> 
> 
> 
> Do you have plans to use `SummarizeContent` in some other place?
> 
> If not, then I think we don't need this component, and we should put it's content back to `SummarizeSidebar`. 
> 
> 
> 
> Hook is good, btw.

@deniskaber sorry, I thought I wrote something in the description. Essentially, I want to use the content from the Summarize component in the SDK and didn't want to have the sidebar styling, so we can have more flexibility on whether we display the summarize component as a popover/drawer/whatever in the SDK

I think it'll make more sense in #44494 when I add the feature to the interactive question

File: frontend/src/metabase/query_builder/components/view/sidebars/SummarizeSidebar/SummarizeContent/SummarizeContent.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    nextAggregation: Lib.Aggregable,
+  ) => void;
+  handleRemoveAggregation: (aggregation: Lib.AggregationClause) => void;
+  handleAddBreakout: (column: Lib.ColumnMetadata) => void;
+  handleUpdateBreakout: (
+    clause: Lib.BreakoutClause,
+    column: Lib.ColumnMetadata,
+  ) => void;
+  handleRemoveBreakout: (column: Lib.ColumnMetadata) => void;
+  handleReplaceBreakouts: (column: Lib.ColumnMetadata) => void;
```
Comment (heypoom) [inline frontend/src/metabase/query_builder/components/view/sidebars/SummarizeSidebar/SummarizeContent/SummarizeContent.tsx:32]: nit: per convention `handle` is used for the handler to pass to the prop, while `on` is the prefix of the prop itself, e.g. `onRemoveBreakout={handleRemoveBreakout}`. can we rename these handle* props to on*?

File: frontend/src/metabase/query_builder/components/view/sidebars/SummarizeSidebar/SummarizeContent/SummarizeContent.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  query,
+  aggregations,
+  hasAggregations,
+  handleAddAggregations,
+  handleUpdateAggregation,
+  handleRemoveAggregation,
+  handleAddBreakout,
+  handleUpdateBreakout,
+  handleRemoveBreakout,
+  handleReplaceBreakouts,
```
Comment (heypoom) [inline frontend/src/metabase/query_builder/components/view/sidebars/SummarizeSidebar/SummarizeContent/SummarizeContent.tsx:45]: nit: you might want to rename this as well, per the above comment.

File: frontend/src/metabase/query_builder/components/view/sidebars/SummarizeSidebar/SummarizeContent/use-summarize-query.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
+import { useCallback, useMemo, useState } from "react";
+
+import * as Lib from "metabase-lib";
+
+export const STAGE_INDEX = -1;
```
Comment (heypoom) [inline frontend/src/metabase/query_builder/components/view/sidebars/SummarizeSidebar/SummarizeContent/use-summarize-query.ts:5]: (I know this is out of scope of this PR, but might be good to add a comment later about what STAGE_INDEX is)

**Review by heypoom — APPROVED** [review]
Per the reply to Denis' comment above, I think extracting the summarization UI outside of the sidebar to use in the SDK makes sense, as we don't need the sidebar in the SDK. I wonder if we can rename `SummarizeContent` to something clearer though.

I also added a small note on the prop names. Thanks!

**Comment by oisincoveney** [comment]
> Per the reply to Denis' comment above, I think extracting the summarization UI outside of the sidebar to use in the SDK makes sense, as we don't need the sidebar in the SDK. I wonder if we can rename `SummarizeContent` to something clearer though.
> 
> I also added a small note on the prop names. Thanks!

I'll make the updates to the prop names for now - we can always change the name of `SummarizeContent` in a future PR

**Pull request merged**